File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/09 Fibonacci Number/fibo.java
class Solution {
    public int fib(int n) {
        // Base cases for Fibonacci series
        if (n == 0) {
            return 0;
        }
        if (n == 1) {
            return 1;
        }

        // Recursive call for Fibonacci
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Example: Get the 10th Fibonacci number
        int result = solution.fib(10);
        System.out.println("The 10th Fibonacci number is: " + result); // Output: 55
    }
}

File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/09 Fibonacci Number/fibo.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        # Base cases for Fibonacci series
        if n == 0:
            return 0
        if n == 1:
            return 1

        # Recursive call for Fibonacci
        return self.fib(n - 1) + self.fib(n - 2)